samples/SimpleDataSource/Tables/Metadata/FileStatsMetadataTable.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Processing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SampleCustomDataSource.Tables.Metadata
+{
+    //
+    // This is a sample Metadata table for .txt files
+    // Metadata tables are used to expose information about the data being processed, not the actual data being processed.
+    // Metadata would be something like "number of events in the file" or "file size" or any other number of things that describes the data being processed.
+    // In this sample table, we expose three columns: File Name, Line Count and Word Count.
+    //
+
+    //
+    // In order for the CustomDataSourceBase to understand your metadata table, 
+    // add a MetadataTable attribute which denotes this table as metadata.
+    //
+
+    [Table]             
+    //
+    // Have the MetadataTable inherit the TableBase class
+    //
+    public sealed class FileStatsMetadataTable
+        : TableBase
+    {
+        public FileStatsMetadataTable(IReadOnlyDictionary<string, IReadOnlyList<Tuple<Timestamp, string>>> lines)
+            : base(lines)
+        {
+        }
+
+        public static TableDescriptor TableDescriptor = new TableDescriptor(
+            Guid.Parse("{40AF86E5-0DF8-47B1-9A01-1D6C3529B75B}"), // The GUID must be unique across all tables
+            "File Stats",                                         // The MetadataTable must have a name
+            "Statistics for text files",                          // The MetadataTable must have a description
+            TableDescriptor.DefaultCategory,
+            true);
+
+        //
+        // Declare columns here. You can do this using the ColumnConfiguration class. 
+        // It is possible to declaratively describe the table configuration as well. Please refer to our Advanced Topics Wiki for more information.
+        //
+        // The Column metadata describes each column in the table. 
+        // Each column must have a unique GUID and a unique name. The GUID must be unique globally; the name only unique within the table.
+        //
+        // The UIHints provides some hints to a viewer on how to render the column. 
```

Comment:
"viewer" is (humorously) ambiguous with the human looking at the table. Perhaps include something like "viewer, such as WPA,"